googlemediaframework/src/main/java/com/google/android/libraries/mediaframework/exoplayerextensions/DashRendererBuilder.java

```diff
@@ -0,0 +1,348 @@
+import com.google.android.exoplayer.dash.mpd.Representation;
+import com.google.android.exoplayer.drm.DrmSessionManager;
+import com.google.android.exoplayer.drm.MediaDrmCallback;
+import com.google.android.exoplayer.drm.StreamingDrmSessionManager;
+import com.google.android.exoplayer.text.TextTrackRenderer;
+import com.google.android.exoplayer.text.ttml.TtmlParser;
+import com.google.android.exoplayer.text.webvtt.WebvttParser;
+import com.google.android.exoplayer.upstream.BufferPool;
+import com.google.android.exoplayer.upstream.DataSource;
+import com.google.android.exoplayer.upstream.DefaultBandwidthMeter;
+import com.google.android.exoplayer.upstream.UriDataSource;
+import com.google.android.exoplayer.util.ManifestFetcher;
+import com.google.android.exoplayer.util.ManifestFetcher.ManifestCallback;
+import com.google.android.exoplayer.util.MimeTypes;
+import com.google.android.exoplayer.util.Util;
+import com.google.android.libraries.mediaframework.exoplayerextensions.ExoplayerWrapper.RendererBuilder;
+import com.google.android.libraries.mediaframework.exoplayerextensions.ExoplayerWrapper.RendererBuilderCallback;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A {@link RendererBuilder} for DASH.
+ */
+public class DashRendererBuilder implements RendererBuilder,
+    ManifestCallback<MediaPresentationDescription> {
+
+  private static final int BUFFER_SEGMENT_SIZE = 64 * 1024;
+  private static final int VIDEO_BUFFER_SEGMENTS = 200;
+  private static final int AUDIO_BUFFER_SEGMENTS = 60;
+  private static final int TEXT_BUFFER_SEGMENTS = 2;
+  private static final int LIVE_EDGE_LATENCY_MS = 30000;
+
+  private static final int SECURITY_LEVEL_UNKNOWN = -1;
+  private static final int SECURITY_LEVEL_1 = 1;
+  private static final int SECURITY_LEVEL_3 = 3;
+
+  private static final String AC_3_CODEC = "ac-3";
+  private static final String E_AC_3_CODEC = "ec-3";
+
+  private final String userAgent;
+  private final String url;
+  private final String contentId;
+  private final MediaDrmCallback drmCallback;
+  private final TextView debugTextView;
+
+  private ExoplayerWrapper player;
+  private RendererBuilderCallback callback;
+  private ManifestFetcher<MediaPresentationDescription> manifestFetcher;
+
+  public DashRendererBuilder(String userAgent, String url, String contentId,
+                             MediaDrmCallback drmCallback, TextView debugTextView) {
+    this.userAgent = userAgent;
+    this.url = url;
+    this.contentId = contentId;
+    this.drmCallback = drmCallback;
+    this.debugTextView = debugTextView;
+  }
+
+  @Override
+  public void buildRenderers(ExoplayerWrapper player, RendererBuilderCallback callback) {
+    this.player = player;
+    this.callback = callback;
+    MediaPresentationDescriptionParser parser = new MediaPresentationDescriptionParser();
+    manifestFetcher = new ManifestFetcher<MediaPresentationDescription>(parser, contentId, url,
```

Comment:
Ok, I don't want to change it unless there's a good reason since this is copied from the ExoPlayer demo (and thus updating it would mean a messier merge the next time they make changes to this code).